pydantic/_internal/_fields.py

```diff
@@ -220,7 +221,7 @@ def collect_fields(  # noqa: C901
                 if not default.init:
                     # dataclasses.Field with init=False are not fields
                     continue
-                if DC_KW_ONLY and default.kw_only is True:
+                if DC_KW_ONLY and default.kw_only is True:  # type: ignore
```

Comment:
what's this type error for?

Reply:
```
Cannot access member "kw_only" for type "Field[Unknown]
```

It's probably because they added it in Python 3.10 or something like that.

Reply:
This is a bit concerning to me, because it makes me think that pyright is treating our `FieldInfo` as a `dataclasses.Field`, due to the `@dataclass_transform` decorator and the `field_specifiers=Field`. I could be wrong about this though.

If I'm right about this, I think this might be one of the more potentially-risky sources of missed type checks. If we continue to run mypy in CI (even in non-recursive mode), this should be okay. (Or at least there's no regression)

Reply:
this is a `dataclasses.Field`, see the `if isinstance(default, dataclasses.Field):` line above.

I think this is fine.